File: codegen/swagger-ehr/dart/lib/model/update_audit.dart
part of swagger.api;

class UpdateAudit {
  
  String type = "UPDATE_AUDIT";

  TerminologyCode changeType = null;

  DvText description = null;

  PartyProxy committer = null;

  UpdateAudit();

  @override
  String toString() {
    return 'UpdateAudit[type=$type, changeType=$changeType, description=$description, committer=$committer, ]';
  }

  UpdateAudit.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    type = json['_type'];
    changeType = new TerminologyCode.fromJson(json['change_type']);
    description = new DvText.fromJson(json['description']);
    committer = new PartyProxy.fromJson(json['committer']);
  }

  Map<String, dynamic> toJson() {
    return {
      '_type': type,
      'change_type': changeType,
      'description': description,
      'committer': committer
     };
  }

  static List<UpdateAudit> listFromJson(List<dynamic> json) {
    return json == null ? new List<UpdateAudit>() : json.map((value) => new UpdateAudit.fromJson(value)).toList();
  }

  static Map<String, UpdateAudit> mapFromJson(Map<String, Map<String, dynamic>> json) {
    var map = new Map<String, UpdateAudit>();
    if (json != null && json.length > 0) {
      json.forEach((String key, Map<String, dynamic> value) => map[key] = new UpdateAudit.fromJson(value));
    }
    return map;
  }
}

File: codegen/oas-ehr/javascript-closure-angular/API/Client/UpdateAudit.js
goog.provide('API.Client.UpdateAudit');

/**
 * The set of attributes required to document the committal of an information item to a repository. Used by the server to create an AUDIT_DETAILS object.
 * @record
 */
API.Client.UpdateAudit = function() {}

/**
 * @type {!string}
 * @export
 */
API.Client.UpdateAudit.prototype.type;

/**
 * @type {!API.Client.TerminologyCode}
 * @export
 */
API.Client.UpdateAudit.prototype.changeType;

/**
 * @type {!API.Client.DvText}
 * @export
 */
API.Client.UpdateAudit.prototype.description;

/**
 * @type {!API.Client.PartyProxy}
 * @export
 */
API.Client.UpdateAudit.prototype.committer;

